### ISSUE TITLE
.net provider 1.8.0-preview throws on add dns record

### ISSUE THREAD

**Issue by juchom** [pulumi/pulumi-cloudflare#62]
I'm trying to use the provider to create a simple DNS record with the following code

```csharp
var dnsRecord = new Record("showcase-web-cname", new RecordArgs
{
    Name = "mywebapp",
    ZoneId = "myzone.tld",
    Type = "CNAME",
    Value = "mywebapp.tld",
    Ttl = 1 // Auto
});
```

And I have this exception on pulumi preview

```
    error: Running program 'C:\dev\myapp\iac\bin\Debug\netcoreapp3.1\myapp.dll' failed with an unhandled exception:                                                                          
    System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.                                                                                         
     ---> System.MissingMethodException: Method not found: 'Void Pulumi.CustomResource..ctor(System.String, System.String, Pulumi.ResourceArgs, Pulumi.ResourceOptions)'.                          
       at Pulumi.Cloudflare.Record..ctor(String name, RecordArgs args, CustomResourceOptions options)                                                                                              
       at myapp.MyStack..ctor() in C:\dev\myapp\iac\MyStack.cs:line 11                                                                                                                 
       --- End of inner exception stack trace ---                                                                                                                                                  
       at System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean wrapExceptions, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor, Boolean& hasNoDefaultCtor)
       at System.RuntimeType.CreateInstanceDefaultCtorSlow(Boolean publicOnly, Boolean wrapExceptions, Boolean fillCache)                                                                          
       at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, Boolean wrapExceptions)                                                       
       at System.Activator.CreateInstance[T]()                                                                                                                                                     
       at Pulumi.Deployment.Runner.RunAsync[TStack]()                                                                                                                                              
                                                                                                                                                                                                   
```

**Comment by mikhailshilkov** [comment]
Hi @juchom Thanks for reporting! There is an issue with `1.8.0` release of this provider, we'll make a patch version. For now, you could fix the problem by adding an explicit dependency to the version `1.12.1` of the `Pulumi` NuGet package to your project file.

**Comment by juchom** [comment]
Hi @mikhailshilkov, thanks for the quick reply but it doesn't work, same error:

Here is my csproj:

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <Nullable>enable</Nullable>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Pulumi" Version="1.12.1-preview"/>
    <PackageReference Include="Pulumi.Azure" Version="2.1.0-preview"/>
    <PackageReference Include="Pulumi.Cloudflare" Version="1.8.0-preview"/>
  </ItemGroup>
</Project>
```

**Comment by juchom** [comment]
@stack72 are you sure this fix works, look at my csproj, I'm pointing to version 1.12.1 and I tried with 1.13.0 and I still have the issue...

**Comment by stack72** [comment]
@juchom I believe the issue is because the Pulumi.Cloudflare package references an older version of Pulumi - you can try the fix above before the official release using this build

https://www.nuget.org/packages/Pulumi.Cloudflare/1.9.0-preview-alpha.1585062637

it will be a patch release and become 1.8.1 when i make the final release

**Comment by juchom** [comment]
I've just tried this

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <Nullable>enable</Nullable>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Pulumi" Version="1.12.1-preview"/>
    <PackageReference Include="Pulumi.Azure" Version="2.1.0-preview"/>
    <PackageReference Include="Pulumi.Cloudflare" Version="1.9.0-preview-alpha.1585062637"/>
  </ItemGroup>
</Project>
```

Issue is still here...

**Comment by juchom** [comment]
Ok this works:

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <Nullable>enable</Nullable>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Pulumi.Azure" Version="2.1.0-preview"/>
    <PackageReference Include="Pulumi.Cloudflare" Version="1.9.0-preview-alpha.1585063609"/>
  </ItemGroup>
</Project>
```

Thanks

**Comment by stack72** [comment]
@juchom My bad - I gave you the wrong master version - I can see that one above s still Pulumi 1.9.1 in its dependency tree

Can you try 1 more for me - I am seeing better luck with this one 

https://www.nuget.org/packages/Pulumi.Cloudflare/1.9.0-preview-alpha.1585063609

Paul

**Comment by stack72** [comment]
ah there we go - it was my fault with the version string. Release 1.8.1 is on its way now that this master version works for you

**Comment by stack72** [comment]
Closed via patch release v1.8.1
